packages/yak-swc/yak_swc/src/lib.rs

```diff
@@ -0,0 +1,741 @@
+use css_in_js_parser::{parse_css, to_css, CommentStateType};
+use css_in_js_parser::{Declaration, ParserState};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::path::Path;
+use std::vec;
+use swc_core::common::comments::Comment;
+use swc_core::common::comments::Comments;
+use swc_core::common::{Spanned, DUMMY_SP};
+use swc_core::ecma::visit::VisitMutWith;
+use swc_core::ecma::{
+  ast::*,
+  visit::{as_folder, FoldWith, VisitMut},
+};
+use swc_core::plugin::errors::HANDLER;
+use swc_core::plugin::metadata::TransformPluginMetadataContextKind;
+use swc_core::plugin::{plugin_transform, proxies::TransformPluginProgramMetadata};
+use utils::add_suffix_to_expr::add_suffix_to_expr;
+use utils::ast_helper::{member_expr_to_strings, TemplateIterator};
+use utils::encode_module_import::{encode_module_import, is_mixin_expression};
+
+mod variable_visitor;
+use variable_visitor::VariableVisitor;
+mod yak_imports;
+use yak_imports::YakImportVisitor;
+
+mod utils {
+  pub mod add_suffix_to_expr;
+  pub mod assert_css_expr;
+  pub mod ast_helper;
+  pub mod encode_module_import;
+  pub mod murmur_hash;
+}
+mod naming_convention;
+use naming_convention::NamingConvention;
+
+mod yak_transforms;
+use yak_transforms::{
+  TransformCssMixin, TransformKeyframes, TransformNestedCss, TransformStyled, YakCss, YakTransform,
+  YakType,
+};
+
+/// Static plugin configuration.
+#[derive(Default, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+  /// Use Readable CSS Variable Names
+  #[serde(default = "bool::default")]
+  pub dev_mode: bool,
+  /// The hash for a css-variable depends on the file name including createVar().
+  /// To ensure that the hash is consistent accross multiple systems the relative path
+  /// from the base dir to the source file is used.
+  #[serde()]
+  pub base_path: String,
+}
+
+pub struct TransformVisitor<GenericComments>
+where
+  // SWC provides different comment types for production and testing
+  // We need to abstract over this to make the plugin testable
+  // @see https://stackoverflow.com/questions/78709909/injecting-comments-with-a-swc-rust-plugin
+  GenericComments: Comments,
+{
+  /// Last css parser state to contiue parsing the next css code from a quasi
+  /// in the same scope
+  current_css_state: Option<ParserState>,
+  /// All css declarations of the current root css expression
+  current_declaration: Vec<Declaration>,
+  /// e.g Button in const Button = styled.button`color: red;`
+  current_variable_name: Option<String>,
```

Comment:
```suggestion
  current_variable_name: Option<Atom>,
```

Also add `use swc_core::atoms::Atom`.

`Id` or `Atom` is cheap to clone or hash, while `String` unconditionally allocates.

Reply:
Sometimes I need the Ident including the scope number e.g. `foo#0` - my solution is `Atom::from(ident.to_string())` but assume that's wrong - isn't it?

Reply:
Then you should use `swc_core::ecma::ast::Id` instead

Reply:
Oh cool! 
That way it's also way easier to see from typing wether its with or without scope